rustworkx-core/src/centrality.rs

```diff
@@ -1143,3 +1132,420 @@ where
+        }
+    }
+    #[test]
+    fn test_weighted_closeness_small_digraph() {
+        let g = petgraph::graph::DiGraph::<u32, f64>::from_edges([
+            (0, 1, 0.7),
+            (1, 2, 0.2),
+            (2, 3, 0.5),
+        ]);
+        let c = newman_weighted_closeness_centrality(&g, false, |x| *x.weight());
+        let result = [Some(0.0), Some(0.7), Some(0.175), Some(0.172131)];
+
+        for i in 0..4 {
+            assert_almost_equal!(result[i].unwrap(), c[i].unwrap(), 1e-4);
+        }
+    }
+
+    #[test]
+    fn test_weighted_closeness_many_to_one_connected_digraph() {
+        let g = petgraph::graph::DiGraph::<u32, f64>::from_edges([
+            (1, 0, 0.1),
+            (2, 0, 0.1),
+            (3, 0, 0.1),
+            (4, 0, 0.1),
+            (5, 0, 0.1),
+            (6, 0, 0.1),
+            (7, 0, 0.1),
+            (0, 8, 1.0),
+        ]);
+        let c = newman_weighted_closeness_centrality(&g, false, |x| *x.weight());
+        let result = [
+            Some(0.1),
+            Some(0.0),
+            Some(0.0),
+            Some(0.0),
+            Some(0.0),
+            Some(0.0),
+            Some(0.0),
+            Some(0.0),
+            Some(0.10256),
+        ];
+
+        for i in 0..9 {
+            assert_almost_equal!(result[i].unwrap(), c[i].unwrap(), 1e-4);
+        }
+    }
+
+    #[test]
+    fn test_weighted_closeness_many_to_one_connected_ungraph() {
+        let g = petgraph::graph::UnGraph::<u32, f64>::from_edges([
+            (1, 0, 0.1),
+            (2, 0, 0.1),
+            (3, 0, 0.1),
+            (4, 0, 0.1),
+            (5, 0, 0.1),
+            (6, 0, 0.1),
+            (7, 0, 0.1),
+            (0, 8, 1.0),
+        ]);
+        let c = newman_weighted_closeness_centrality(&g, false, |x| *x.weight());
+        let result = [
+            Some(0.112676056),
+            Some(0.056737588),
+            Some(0.056737588),
+            Some(0.056737588),
+            Some(0.056737588),
+            Some(0.056737588),
+            Some(0.056737588),
+            Some(0.056737588),
+            Some(0.102564102),
+        ];
+
+        for i in 0..9 {
+            assert_almost_equal!(result[i].unwrap(), c[i].unwrap(), 1e-4);
+        }
+    }
+
+    #[test]
+    fn test_weighted_closeness_many_to_one_not_connected_2_digraph() {
+        let g = petgraph::graph::DiGraph::<u32, f64>::from_edges([
+            (1, 0, 0.1),
+            (2, 0, 0.1),
+            (3, 0, 0.1),
+            (4, 0, 0.1),
+            (5, 0, 0.1),
+            (6, 0, 0.1),
+            (7, 0, 0.1),
+            (1, 7, 1.0),
+        ]);
+        let c = newman_weighted_closeness_centrality(&g, false, |x| *x.weight());
+        let result = [
+            Some(0.1),
+            Some(0.0),
+            Some(0.0),
+            Some(0.0),
+            Some(0.0),
+            Some(0.0),
+            Some(0.0),
+            Some(1.0),
+        ];
+
+        assert_eq!(result, *c);
+    }
```

Comment:
IMO this test should keep `assert_eq!` predicate but let me know :D

Reply:
Sure, this one is fine